check_notebooks.py

```diff
@@ -0,0 +1,35 @@
+import logging
+from pathlib import Path
+from nbconvert.preprocessors import ExecutePreprocessor, CellExecutionError
+import nbformat
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s %(message)s')
+
+
+KERNEL_NAME = "notebooks-nwp-env"
+ROOT_DIR = Path(__file__).resolve().parent
+NOTEBOOK_DIRS = [ROOT_DIR, ROOT_DIR / "clean_notebooks"]
+
+def run_notebook(notebook_path):
+    with open(notebook_path) as f:
+        nb = nbformat.read(f, as_version=4)
+
+    # configure the notebook execution mode
+    ep = ExecutePreprocessor(timeout=600, kernel_name=KERNEL_NAME)
+
+    try:
+        # run the notebook
+        ep.preprocess(nb, {'metadata': {'path': notebook_path.parent}})
+        logger.info(f"Notebook ran successfully: {notebook_path}")
+    except CellExecutionError as e:
+        logger.error(f"Error in notebook: {notebook_path}")
+        logger.exception(e)
```

Comment:
```suggestion
        logger.exception(f"Error in notebook: {notebook_path}")
        sys.exit(1)
```
or implement a flag that tracks if an error has happened and exit with code 1 at the end if you want to run all notebooks

Reply:
I implemented the flag that tracks an error and exist at the end